### PR TITLE
Optimize static file loading

### PR DESCRIPTION
**Pull request by rtibbles**
### Summary
* Runs collectstatic once from initialize if any updates have occurred.
* Always runs collectstatic with the `clear` argument to clear the files before collecting
* Runs a capability check to see if we can generate symlinks
* If so, runs collectstatic with `link` option to use symlinks instead of copies to reduce file duplication.

### Reviewer guidance
Does everything still seem to work OK? Does everything work OK on Windows?

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/plugins/utils/__init__.py
Comment (indirectlylit) [inline kolibri/plugins/utils/__init__.py:376]: minor stylistic note on code that pre-dated this PR: I usually like to bail early out of functions so the bulk of the logic isn't nested in a conditional:

```python
def run_plugin_updates():
    if not config["UPDATED_PLUGINS"]:
        return False

    logger.info(
        "Detected updates to plugins: {}".format(
            ", ".join(config["UPDATED_PLUGINS"])
        )
    )
    update_manager = PluginUpdateManager(config["UPDATED_PLUGINS"])
    update_manager.update_plugins()
    if update_manager.errors:
        raise PluginUpdateException(
            "{} failed to update, please restart Kolibri".format(
                ", ".join(update_manager.errors)
            )
        )
    return True
```

The same strategy can be used to flatten the function below:

```python
    if skip_update:
        return

    # Run any plugin specific updates here in case they were missed by
    # our Kolibri version based update logic.
    updated = run_plugin_updates() or updated

    if not updated:
        return

    logger.info("Copying updated static files")
    # Collect any static files
    collectstatic_options = {
        "interactive": False,
        "verbosity": 0,
        "clear": True,
    }
    if CAN_USE_SYMLINKS:
        collectstatic_options["link"] = True
    call_command("collectstatic", **collectstatic_options)
```

Comment (rtibbles) [reply kolibri/plugins/utils/__init__.py:376]: I tried to do a minimal diff here to prevent merge conflicts with the changes that have happened in develop. Happy to make further updates once this gets there.

**Comment by indirectlylit** [comment]
@mrpau-dev would you mind testing this PR?

**Comment by cpauya** [comment]
All good so far!  Tested these on Firefox v75.0 under Win7 VM:

* Import content
    * radina channel
    * TESSA
* Users
    * Class, assign coaches, enroll learners
* Coach class
    * plan topics, plan quiz
* Learn
    * Class (Lesson, Quiz), Channels, Take quiz
Screen Shot 2020-04-29 at 5 34 01 PM


Or are there other features we should test in particular?

**Review by indirectlylit — APPROVED** [review]
that works for me if it sounds like sufficient testing for you @rtibbles

**Comment by rtibbles** [comment]
I think the only other thing we'd want to check is that the symlinking of files doesn't mess with nginx in kolibri-server, but this is probably best done as a QA pass after merge?

**Comment by indirectlylit** [comment]
perhaps @jredrejo should review?

**Comment by jamalex** [comment]
Main thing I raised that would be good to test is switching KOLIBRI_HOME to be on another filesystem (than where the Kolibri Python package has been installed), especially if that's a different filesystem type (e.g. FAT/NTFS) and making sure that works too.

**Comment by indirectlylit** [comment]
thanks - tracked that task here: https://github.com/learningequality/kolibri/issues/6839

**Pull request merged**
